experiments/dataSources/NPMSearch/readme.md

```diff
@@ -0,0 +1,77 @@
+- Searching / filtering for a particular record.
+- Dispaly of detail information (and related lists) about a single record.
+
+## How to use this data source
+
+### Create the data source
+
+First you'll need to create a new REST data source. 
+* Type: REST
+* Name: NPM
+* URL / Endpoint: ``https://api.npms.io/v2``
+* Use Proxy: False
+* Authentication Method: No Authentication
+
+
+#### Double check the remote site setting
+
+* If you are using Skuid on Salesforce, we will automatically set up a remote site setting _for your org_  for this data source. If your pages do not work, you might need to edit it.  Check in Salesforce Setup at **Security > Remote site settings**.   
+
+#### Create a CORS entry
+
+* Allowing the browser to retrieve data directly is the fastest method for data sources (as opposed to using the Server Proxy), but if you are using Skuid on Salesforce, you will be required that you whitelist the endpoint so you don't have a CORS violation.
+* Within Salesforce Setup, go to **Security > CORS**.
+* Create a new entry - generally it will be the same domain where you ultimately intend to host the page. ``https://<<mydomain>>.visual.force.com``
+
+### Create a page with this REST model to search for NPM Packages
+
+- Create a new page
+
+- Create a model that searches the NPM registry for particular keywords. 
+    - Data Source Type: REST
+    - Data Source: NPM
+    - Model Behavior: Read-Only
+    - Data Source URL: /search?q={{search}}
+    - Query on page load: Off. Since this is a Search model - it will need a keyword passed into the merge variable {{search}} before a query is executed, or you will get an error.
+    - All other properties can be default. 
+
+- Select “Path to Contents” and choose “Results”.  
+
+- Select "Fields" on the model properties to see the available fields. You can traverse down through the data structure to select specific fields to include in your model. 
+
+- Create a condition that interacts with the {{search}} parameter 
+    - Type: Url Parameter
+    - URL Parameter: Search
+    - Value: Leave this blank
+    - State: Filterable Default off. 
+
+## Page Ideas
+
+- Add a table to search for NPM Packages and display information about them. 
+- Turn on the search control and have it interact with the model condition you created (Server search). 
+- It will also be important for the “Empty Search Behavior” property to be set to “Remove all Model Data”. If you use the "Re-query" property - a blank search will be processed which will cause an error.
+
+## Sample Page
+
+Create a new V2 Page and copy the XML from "NPMS_Package_Search" file in this directory to see a more fully constructed package search tool. It performs the package search we created above, and then lets you select from the search results to retrieve an interesting amount of analysis about a single NPM Package.
```

Comment:
So after creating this sample page, Skuid's App Composer prompts you for sample values for metadata retrieval. Clicking cancel on all of these works, but presents a page problem.  Can we note this in the instructions? Or provide a sample value in these instructions? I tried a few things but kept getting an `Endpoint does not exist` error.

This was also the point where I was prompted to create a remote site setting, so might note that too?